lib/light.js

```diff
@@ -46,9 +48,68 @@ function clampColorTemp(colorTemp, colorTempMin, colorTempMax, logger) {
     return colorTemp;
 }
 
+/**
+ * Converts converter's value input to {x,y}, {h,s,v}, {h,s}, {h} or {s}
+ * @param {object} value converter value parameter
+ * @return {object} normalized color information
+ */
+function extractColorFromValue(value) {
+    if (value.hasOwnProperty('x') && value.hasOwnProperty('y')) {
+        return value;
+    } else if (value.hasOwnProperty('r') && value.hasOwnProperty('g') && value.hasOwnProperty('b')) {
+        return utils.rgbToXY(value.r, value.g, value.b);
+    } else if (value.hasOwnProperty('rgb')) {
+        return utils.rgbToXY(...value.rgb.split(',').map((i) => parseInt(i)));
+    } else if (value.hasOwnProperty('hex') || (typeof value === 'string' && value.startsWith('#'))) {
+        return utils.hexToXY(typeof value === 'string' && value.startsWith('#') ? value : value.hex);
+    } else if (value.hasOwnProperty('h') && value.hasOwnProperty('s') && value.hasOwnProperty('l')) {
+        return utils.hslToHSV(value.h, value.s, value.l);
+    } else if (value.hasOwnProperty('hsl')) {
+        return utils.hslToHSV(...value.hsl.split(',').map((i) => parseInt(i)));
+    } else if (value.hasOwnProperty('h') && value.hasOwnProperty('s') && value.hasOwnProperty('b')) {
+        return {h: value.h, s: value.s, v: value.b};
+    } else if (value.hasOwnProperty('hsb')) {
+        const hsv = value.hsb.split(',').map((i) => parseInt(i));
+        return {h: hsv[0], s: hsv[1], v: hsv[2]};
+    } else if (value.hasOwnProperty('h') && value.hasOwnProperty('s') && value.hasOwnProperty('v')) {
+        return value;
```

Comment:
nit: return these as 'hue', 'saturation', and 'brightness' to stick close to ZCL (if v is 0-100, we can map it to 0-254)